Windows Clang Build Error

Commit 2f5ae3dc2aeab8c296aa5710975cbcad3e56f098 broke clang on Windows. The `::` global namespace prefix of `::stderr` isn't compatible with the macro. [C++ reference](https://en.cppreference.com/w/cpp/io/c/std_streams) says it should be a macro, so we probably need to remove the `::`

```sh
.../build/_deps/mdspan-src/include\mdspan\../experimental/__p0009_bits\macros.hpp:127:18: error: expected unqualified-id
  127 |   std::fprintf(::stderr, "%s:%u: precondition failure: `%s`\n", file, line, cond);

      |                  ^

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h:38:16: note: expanded from macro 'stderr'

   38 | #define stderr (__acrt_iob_func(2))

      |                ^

1 error generated.
```

--- a/include/experimental/__p0009_bits/macros.hpp
+++ b/include/experimental/__p0009_bits/macros.hpp
@@ -124,7 +124,7 @@ MDSPAN_FUNCTION inline void default_precondition_violation_handler(const char* c
 #else
 MDSPAN_FUNCTION inline void default_precondition_violation_handler(const char* cond, const char* file, unsigned line)
 {
-  std::fprintf(::stderr, "%s:%u: precondition failure: `%s`\n", file, line, cond);
+  std::fprintf(stderr, "%s:%u: precondition failure: `%s`\n", file, line, cond);
   std::abort();
 }
 #endif

